x-pack/plugins/serverless_security/public/components/get_started/helpers.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getSections } from './sections';
+import { ActiveCard, CardId, ProductId, SectionId, StepId } from './types';
+
+export const setupCards = (
+  finishedSteps: Record<CardId, Set<StepId>>,
+  activeProducts: Set<ProductId>
+) =>
+  activeProducts.size > 0
+    ? getSections().reduce((acc, section) => {
+        const cardsInSections = section.cards?.reduce((accCards, card) => {
+          if (
+            !card.productTypeRequired ||
+            card?.productTypeRequired?.some((condition) => activeProducts.has(condition))
+          ) {
+            const stepsDone = finishedSteps[card.id] ?? new Set();
+            const timeInMins =
+              card?.steps?.reduce(
+                (totalMin, { timeInMinutes, id: stepId }) =>
+                  (totalMin += stepsDone.has(stepId) ? 0 : timeInMinutes ?? 0),
+                0
+              ) ?? 0;
+            const stepsLeft = (card?.steps?.length ?? 0) - (stepsDone?.size ?? 0);
```

Comment:
NIT: I see the "is active" condition, the `timeInMins` and `stepsLeft` logic is repeated in the `updateCard` function.
Could we extract this logic into functions so we do not duplicate code, it will be more readable as well:

```suggestion
          if (isCardActive(card, activeProducts)) {
            const stepsDone: Set<StepId> = finishedSteps[card.id] ?? new Set();
            const timeInMins = getCardTimeInMinutes(card, stepsDone);
            const stepsLeft = getCardStepsLeft(card, stepsDone);
```